test/validate.spec.js

```diff
@@ -45,31 +46,46 @@ describe('lib/validate.js', () => {
 	});
 
 	describe('validator function', () => {
-		it('returns false if publicKey is falsy', () => {
-			keyFactory.returns(false);
+		['', false, undefined, null].forEach((value) => {
+			it(`returns false if publicKey is ${value}`, () => {
+				keyFactory.returns(value);
+
+				const validator = validatorFactory(keyFactory);
+				const result = validator('test-key', 'test-token');
+
+				keyFactory.should.have.been.calledOnce;
+				result.should.equal(false);
+			});
+		});
+
+		it('returns false if the supplied key is empty', () => {
+			const token = sign('', key)
 
 			const validator = validatorFactory(keyFactory);
-			const result = validator('test-key', 'test-token');
+			const result = validator('charlie.briggs', token);
 
 			keyFactory.should.have.been.calledOnce;
 			result.should.equal(false);
 		});
 
-		it('returns true if supplied key validates against public key', () => {
-			keyFactory.returns('test-public-key');
+		it('returns false if supplied key does not validates against public key', () => {
+			const token = sign('charlie.briggs-lantern.ft.com', key)
 
 			const validator = validatorFactory(keyFactory);
-			const result = validator('test-key', 'test-token');
-			const testBuffer = new Buffer('test-public-key', 'base64');
+			const result = validator('charlie.briggs', token);
 
-      // This should be improved at some point.
-			result.should.equal(true);
 			keyFactory.should.have.been.calledOnce;
-			rsaStub.withArgs(testBuffer).should.have.been.calledOnce;
-			exportKeyStub.should.have.been.calledOnce;
-			createVerifyStub.should.have.been.calledOnce;
-			updateSpy.withArgs('test-key').should.have.been.calledOnce;
-			verifyStub.withArgs('test-pem', 'test-token').should.have.been.calledOnce;
+			result.should.equal(false);
+		});
+
+		it('returns true if supplied key validates against the public key', () => {
+			const token = sign('charlie.briggs:lantern.ft.com', key)
```

Comment:
Why si the token coon separated, but the token in the previous test hyphen separated?

Reply:
Oops, last minute change